sway-ir/src/instruction.rs

```diff
@@ -181,6 +195,7 @@ impl Instruction {
         match self {
             Instruction::AddrOf(_) => Some(Type::Uint(64)),
             Instruction::AsmBlock(asm_block, _) => Some(asm_block.get_type(context)),
+            Instruction::BinaryOp { arg1, .. } => arg1.get_type(context),
```

Comment:
We might want to add a return type to the `BinaryOp` struct for the future, as the assumption that both args have the same return type as the expression won't necessarily be true always.  But not in this PR...